ocf_datapipes/select/select_time_periods.py

```diff
@@ -0,0 +1,25 @@
+from typing import Union
+
+import pandas as pd
+import xarray as xr
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+
+@functional_datapipe("select_time_periods")
+class SelectTimePeriodsIterDataPipe(IterDataPipe):
+    def __init__(
+        self, source_dp: IterDataPipe, time_periods: pd.DataFrame, dim_name: str = "time_utc"
```

Comment:
can we call this `datetime_utc`, 'time' to me means HH:MM:SS, so i get confused about this